document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoQueryParser.java

```diff
@@ -0,0 +1,103 @@
+package org.hypertrace.core.documentstore.mongo;
+
+import static org.hypertrace.core.documentstore.Collection.UNSUPPORTED_QUERY_OPERATION;
+
+import com.mongodb.BasicDBObject;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.hypertrace.core.documentstore.Filter;
+import org.hypertrace.core.documentstore.OrderBy;
+
+class MongoQueryParser {
+
+  static Map<String, Object> parseFilter(Filter filter) {
+    if (filter.isComposite()) {
+      Filter.Op op = filter.getOp();
+      switch (op) {
+        case OR:
+        case AND:
+          {
+            List<Map<String, Object>> childMapList =
+                Arrays.stream(filter.getChildFilters())
+                    .map(MongoQueryParser::parseFilter)
+                    .filter(map -> !map.isEmpty())
+                    .collect(Collectors.toList());
+            if (!childMapList.isEmpty()) {
+              return Map.of("$" + op.name().toLowerCase(), childMapList);
+            } else {
+              return Collections.emptyMap();
+            }
+          }
+        default:
+          throw new UnsupportedOperationException(
+              String.format("Boolean operation:%s not supported", op));
+      }
+    } else {
+      Filter.Op op = filter.getOp();
+      Object value = filter.getValue();
+      Map<String, Object> map = new HashMap<>();
+      switch (op) {
+        case EQ:
+          map.put(filter.getFieldName(), value);
```

Comment:
Looks like this just got moved. The create/put/break/return kills me but that's probably scope creep.